File: java/altgraph_data_app/src/main/java/org/cjoakim/cosmos/altgraph/data/common/model/npm/Author.java
package org.cjoakim.cosmos.altgraph.data.common.model.npm;

import com.azure.spring.data.cosmos.core.mapping.Container;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.Data;
import lombok.EqualsAndHashCode;

/**
 * Instances of this class represent a NPM (Node.js Package Manager) library Author.
 * <p>
 * Chris Joakim, Microsoft, November 2022
 */

@Data
@EqualsAndHashCode(callSuper = false)
@Container(containerName = "npm_graph")
@JsonIgnoreProperties(ignoreUnknown = true)
public class Author extends NpmDocument {

    private String authorName;

    public Author() {

        super();
    }

    @Override
    public void intern() {

        baseIntern();
    }
}

File: dotnet/src/altgraph_shared_app/Models/Npm/Author.cs
using System.Text.Json.Serialization;
using Microsoft.Azure.CosmosRepository.Attributes;

namespace altgraph_shared_app.Models.Npm
{
  public class Author : NpmDocument
  {
    [JsonPropertyName("authorName")]
    public string AuthorName { get; set; } = string.Empty;
  }
}